docs/DX_IMPROVEMENTS.md

```
@@ -0,0 +1,239 @@
+# Developer Experience Improvements from microsoft/typescript-go
+
+This document tracks the DX improvements adopted from [microsoft/typescript-go](https://github.com/microsoft/typescript-go).
+
+## Implementation Status
+
+| Feature | Status | Priority | Notes |
+|---------|--------|----------|-------|
```

markdownlint-cli2 (0.20.0)

[warning] Table column style
Table pipe is missing space to the right for style "compact"

(MD060, table-column-style)